RocketChat/Rocket.Chat.ReactNative
# Chore: Update Storybook libs to v6

## Proposed changes

- [x] Upgrades Storybook to 6.3
  - https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#from-version-53x-to-60x
  - https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#from-version-60x-to-610
  - https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#from-version-61x-to-620
  - https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#from-version-62x-to-630
- [x] Since I had to rewrite those tests, I already migrated them to Typescript (also migrated a few small files I had to touch)
- [x] I had to migrate from @react-native-community/async-storage to @react-native-async-storage/async-storage

## Issue(s)	


## How to test or reproduce

- Unit and E2E tests should be passing
- ScreenLock should keep working
  - need to test app version update... it should be just a reorg change, but we need to test it anyway

## Screenshots

## Types of changes




## Checklist



## Further comments
